sdk/storage/storage-queue/RecordAndPlayback.md

```diff
@@ -0,0 +1,199 @@
+  - **Progress:** Nock does not record a request if it's aborted in a 'progress' callback
+  - **Size:** the generated recording file is too big and would considerably increase the size of the package
+  - **Tempfile:** the request makes use of a random tempfile created locally, and the recorder does not support recording it as unique information
+  - **UUID:** a UUID is randomly generated within the SDK and used in an HTTP request, resulting in Nock being unable to recognize it
+
+- We leverage mocha's `.skip()` functionality to skip the test
+  `this.skip()` - https://mochajs.org/#inclusive-tests
+  By this, the tests in the skip list will only be executed if the `TEST_MODE` is neither `"record"` nor `"playback"`.
+- Number of tests being covered in storage-queue:
+  - Node: 48/48 (100%)
+  - Browser: 30/35 (85%)
+
+---
+
+## Recordings
+
+- Recordings are being saved in one folder for each describe-block test suite
+  recording file structure
+  - `recordings/node/<describe-block-title>/recording_<test-title>.js` for node tests and
+  - `recordings/browsers/<describe-block-title>/recording_<test-title>.json` for browser tests.
+- The file name of the recording for a test preserves the title of the `describe`-block and the corresponding `it`-block (with special characters appropriated). The recordings for browser tests go into the `./recordings/browsers/` folder (Similarly, node recordings in the `./recordings/node/` folder).
+
+  For example
+
+  ```typescript
+  describe("Aborter", () => {
+    it("should abort when abort() is called", async () => {
+      ...
+      ...
+    });
+  });
+  ```
+
+  (Browser Test) Recording corresponding to the above test is placed at
+  `./recordings/browsers/aborter/recording_should_abort_when_abort_is_called.json`
+
+  [ Following this rule - `./recordings/browsers/<describe-block-title>/recording_<test-title>.json` ]
+
+- Just like the test recordings, requests and responses from the `before` and `after` sections of the `describe` block are saved at - `recordings/{node|browsers}/<describe-block-title>/recording_{before|after}_all_hook.{js|json}`.
+
+---
+
+## Updating an existing test/test-suite
+
+- Recordings are saved in `./recordings/` folder.
+- If a test/test-suite is modified, we might need to record it again in order to equip the recording to accommodate any updates in the function calls that invoke http requests.
+
+  - execute the test/test-suite script(or all the tests) by setting the env variable `TEST_MODE = record`
+
+- If only the test title is updated,
+
+  - we can either execute that specific test again to generate the recording
+
+    or
+
+  - we can just update the file name of the recording accordingly
+
+- **Commit** the generated recordings.
+
+---
+
+## Adding a new test/test-suite
+
+- Any function call that is not in the `it`-block and belongs to a `describe`-block must go in one of the `beforeEach`, `afterEach`, `before` or `after` sections.
```

Comment:
We should be allowed to make function calls in `describe` as long as it doesn't send HTTP requests and it doesn't generate unique info we should store.